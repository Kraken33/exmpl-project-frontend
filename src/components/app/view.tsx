import { APP_NAME } from "consts";
import React from "react";
import Helmet from "react-helmet";
import { Switch } from "react-router-dom";
import { routers } from "routes";

const View: React.FC = () => {
  return (
    <>
      <Helmet defaultTitle={`${APP_NAME}`} titleTemplate={`%s | ${APP_NAME}`} />
      <Switch>{routers}</Switch>
    </>
  );
};

export { View };
