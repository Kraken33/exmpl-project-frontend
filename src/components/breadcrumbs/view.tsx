import { Breadcrumb } from "antd";
import React from "react";
import { Link } from "react-router-dom";

import { BreadcrumbComponent } from "./types";

const View: React.FC<BreadcrumbComponent> = ({ breadcrumbs }) => {
  const isLast = (index: number) => breadcrumbs.length - 1 !== index;

  return (
    <Breadcrumb className={"breadcrumbs"}>
      {breadcrumbs?.map(({ breadcrumb, match }, index) => {
        return (
          <Breadcrumb.Item key={breadcrumb.key}>
            {!isLast(index) ? (
              <Link to={match.url}>{breadcrumb}</Link>
            ) : (
              <span>{breadcrumb}</span>
            )}
          </Breadcrumb.Item>
        );
      })}
    </Breadcrumb>
  );
};

export { View };
