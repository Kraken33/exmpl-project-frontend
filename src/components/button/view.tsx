import { LoadingOutlined } from "@ant-design/icons";
import { Button } from "antd";
import React from "react";

import { ButtonProps } from "./types";

const Component: React.FC<ButtonProps> = ({ pending, children, ...rest }) => {
  const props = {
    ...rest,
    ...(pending
      ? { icon: pending ? <LoadingOutlined /> : undefined, disabled: true }
      : {}),
  };
  return <Button {...props}>{children}</Button>;
};

export { Component };
