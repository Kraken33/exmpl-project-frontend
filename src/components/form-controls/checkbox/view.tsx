import { Checkbox } from "antd";
import FormItem from "components/form-item";
import React from "react";

import { FControlCheckbox } from "./types";

const Component: React.FC<FControlCheckbox> = ({
  input,
  meta,
  type,
  label,
  required,
  ...attributes
}) => {
  return (
    <FormItem required={required} label={label} {...meta}>
      <Checkbox
        onChange={input.onChange}
        checked={input.value}
        {...attributes}
      />
    </FormItem>
  );
};

export { Component };
