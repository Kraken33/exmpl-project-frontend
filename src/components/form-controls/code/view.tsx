import React from 'react';
import { Controlled as CodeMirror } from "react-codemirror2";

import { FControlCode } from './types';

import FormItem from 'components/form-item';

const Component: React.FC<FControlCode> = ({ input, meta, type, label, required, ...attributes }) => {
  return (
    <FormItem required={required} label={label} {...meta}>
      <CodeMirror
        {...attributes}
        options={{
          theme: 'material',
          lineNumbers: true
        }}
        value={input.value}
        onBeforeChange={(_, __, value) => {
          input.onChange(value);
        }}
      />
    </FormItem>
  );
}

export { Component };