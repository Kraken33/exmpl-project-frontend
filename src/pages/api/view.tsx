import { Button, Col, Divider, Row, Select, Tabs, TreeSelect } from "antd";
import { EMPTY_STRING, HttpMethods } from "consts";
import { map } from "lodash/fp";
import React, { useState } from "react";
import { Controlled as CodeMirror } from "react-codemirror2";
import ReactJson from "react-json-view";
import { http } from "services/http";

import styles from "./index.module.scss";
import {
  EKeys,
  ERequestDataType,
  InnerProps,
  KeysUnion,
  RequestDataTypes,
  State,
} from "./types";

const { TreeNode } = TreeSelect;
const { TabPane } = Tabs;

const Component: React.FC<InnerProps> = ({ routes, intl }) => {
  const methods: Array<keyof typeof HttpMethods> = [
    HttpMethods.POST,
    HttpMethods.GET,
    HttpMethods.PUT,
    HttpMethods.DELETE,
  ];

  const [state, setState] = useState<State>({
    method: methods[0],
    request: EMPTY_STRING,
    json: EMPTY_STRING,
    header: EMPTY_STRING,
    response: {},
  });

  const {
    apiRequestPlaceholder,
    apiSubmitButtonLabel,
    apiRequestDataTypeJson,
    apiRequestDataTypeHeader,
  } = intl;

  const requestDataTypes: RequestDataTypes = [
    {
      value: ERequestDataType.json,
      label: apiRequestDataTypeJson,
    },
    {
      value: ERequestDataType.header,
      label: apiRequestDataTypeHeader,
    },
  ];
  const isSubmitted = !!state.request;

  const getStateProp = (name: KeysUnion): { value: string; onChange: any } => ({
    value: state[name],
    onChange: (e: any) => {
      setState({
        ...state,
        [name]: e,
      });
    },
  });

  const getCodeMirrorProp = (
    name: KeysUnion
  ): { value: string; options: any; onBeforeChange: any } => ({
    options: {
      theme: "material",
    },
    value: state[name],
    onBeforeChange: (_: any, __: any, value: string) => {
      setState({ ...state, [name]: value });
    },
  });

  const sendRequest = async () => {
    try {
      const { method, request } = state;
      const json = state.json ? JSON.parse(state.json) : {};
      // @TODO Finish it (data в зависимости от метода и хедеры подключить)

      const response = await http({
        url: request,
        method: method as any,
        params: json,
        data: json,
        headers: {},
      });

      setState({ ...state, response });
    } catch (e) {
      setState({ ...state, response: e });
    }
  };

  const renderRequestDataTypes = map(({ value, label }) => (
    <TabPane tab={label} key={value}>
      <CodeMirror {...getCodeMirrorProp(value)} />
    </TabPane>
  ));

  const renderMethodOptions = map((method: string) => (
    <Select.Option key={method} value={method}>
      {method}
    </Select.Option>
  ));

  return (
    <div className={styles.api}>
      <Row>
        <Col span={24}>
          <Row>
            <Col span={6}>
              <Select
                {...getStateProp(EKeys.method)}
                className={styles.api_methods}
              >
                {renderMethodOptions(methods)}
              </Select>
            </Col>
            <Col span={14}>
              <TreeSelect
                showSearch
                dropdownStyle={{ maxHeight: 400, overflow: "auto" }}
                placeholder={apiRequestPlaceholder}
                allowClear
                className={styles.api_request}
                treeDefaultExpandAll
                {...getStateProp(EKeys.request)}
              >
                {routes.map((route: any) => (
                  <TreeNode value={route()} title={route()} key={route()} />
                ))}
              </TreeSelect>
            </Col>
            <Col span={4}>
              <Button
                className={styles.api_submit}
                disabled={!isSubmitted}
                onClick={sendRequest}
              >
                {apiSubmitButtonLabel}
              </Button>
            </Col>
          </Row>
        </Col>
        <Col span={24}>
          <Tabs defaultActiveKey={requestDataTypes[0].value as any}>
            {renderRequestDataTypes(requestDataTypes)}
          </Tabs>
        </Col>
        <Col span={24}>
          <Divider />
          <ReactJson src={state.response} />
        </Col>
      </Row>
    </div>
  );
};

export { Component };
