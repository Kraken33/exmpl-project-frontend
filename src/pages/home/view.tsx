import React from "react";

import { useIntl } from 'services/intl';

const HomeComponent: React.FC<unknown> = () => {
  const { test } = useIntl();

return <div>React Typescript boilerplate {test}</div>
};

export const View = HomeComponent;
