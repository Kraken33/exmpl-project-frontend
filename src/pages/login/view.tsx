import { LockOutlined, UserOutlined } from "@ant-design/icons";
import { Button, Card, Form } from "antd";
import logo from "assets/images/logo-dark.svg";
import { FormControl, FormControlTypes } from "components/form-controls";
import React from "react";
import { Link } from "react-router-dom";
import { AvailableRoutes, getLink } from "routes";

import { FormItemNames } from "./consts";
import styles from "./index.module.scss";
import { InnerProps } from "./types";

const Component: React.FC<InnerProps> = ({
  intl,
  authenticate,
  handleSubmit,
}) => {
  return (
    <div className={styles.login}>
      <Card className={styles.login_container}>
        <div className={styles.login_logo}>
          <img src={logo} alt="logo" />
        </div>
        <Form onFinish={handleSubmit(authenticate)}>
          <FormControl
            name={FormItemNames.login}
            type={FormControlTypes.input}
            prefix={<UserOutlined />}
            placeholder={intl.loginLoginFieldPlaceholder}
          />
          <FormControl
            name={FormItemNames.password}
            type={FormControlTypes.password}
            prefix={<LockOutlined />}
            placeholder={intl.loginPasswordFieldPlaceholder}
          />
          <FormControl
            name={FormItemNames.remember}
            type={FormControlTypes.checkbox}
          >
            {" "}
            {intl.loginRememberMe}
          </FormControl>

          <Form.Item>
            <Button type="primary" htmlType="submit">
              {intl.loginButtonSubmit}
            </Button>
            <Link to={getLink(AvailableRoutes.home)}>
              {" "}
              {intl.loginRegistrNow}
            </Link>
          </Form.Item>
        </Form>
      </Card>
    </div>
  );
};

export { Component };
