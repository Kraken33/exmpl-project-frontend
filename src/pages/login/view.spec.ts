import { shallow } from "enzyme";
import { r } from "utils/component-test";

import { Component } from "./view";

describe("testing login page", () => {
  const handleSubmit = jest.fn();
  const props = { handleSubmit, intl: {} };

  it("should render without errors", () => {
    const wrapper = shallow(r(Component, props));
    expect(wrapper).toBeTruthy();
  });

  it("match snapshot", () => {
    const wrapper = shallow(r(Component, props));
    expect(wrapper).toMatchSnapshot();
  });
});
