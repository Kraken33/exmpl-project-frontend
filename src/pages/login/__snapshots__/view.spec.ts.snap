// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing login page match snapshot 1`] = `
<div
  className="login"
>
  <Card
    className="login_container"
  >
    <div
      className="login_logo"
    >
      <img
        alt="logo"
        src="logo-dark.svg"
      />
    </div>
    <ForwardRef(InternalForm)>
      <Component
        name="login"
        prefix={<ForwardRef(UserOutlined) />}
        type="input"
      />
      <Component
        name="password"
        prefix={<ForwardRef(LockOutlined) />}
        type="password"
      />
      <Component
        name="remember"
        type="checkbox"
      >
         
      </Component>
      <FormItem>
        <Connect(Component)
          htmlType="submit"
          on="@LOGIN/AUTHENTICATE_REQUEST"
          type="primary"
        />
        <Link>
           
        </Link>
      </FormItem>
    </ForwardRef(InternalForm)>
  </Card>
</div>
`;
